internal/future/serving/server.go

```diff
@@ -90,3 +129,78 @@ func (p *Params) AddHandleFunc(handlerFunc GrpcHandler, grpcProxyHandler GrpcPro
 	}
 	return p
 }
+
+// invalidate closes all the TCP listeners that would otherwise leak if initialization fails.
+func (p *Params) invalidate() {
+	if err := p.grpcListener.Close(); err != nil {
+		serverLogger.Errorf("error closing grpc handler, %s", err)
+	}
+	if err := p.grpcProxyListener.Close(); err != nil {
+		serverLogger.Errorf("error closing grpc-proxy handler, %s", err)
+	}
+}
+
+// Server hosts a gRPC and HTTP server.
+// All HTTP traffic is served from a common http.ServeMux.
+type Server struct {
+	serverWithProxy grpcServerWithProxy
+}
+
+// grpcServerWithProxy this will go away when insecure.go and tls.go are merged into the same server.
+type grpcServerWithProxy interface {
+	start(*Params) (func(), error)
+	stop()
+}
+
+// Start the gRPC+HTTP(s) REST server.
+func (s *Server) Start(p *Params) (func(), error) {
+	if p.usingTLS() {
+		s.serverWithProxy = newTLSServer(p.grpcListener, p.grpcProxyListener)
+	} else {
+		s.serverWithProxy = newInsecureServer(p.grpcListener, p.grpcProxyListener)
+	}
+	return s.serverWithProxy.start(p)
+}
+
+// Stop the gRPC+HTTP(s) REST server.
+func (s *Server) Stop() {
+	s.serverWithProxy.stop()
+}
+
+// New creates an instance of an Open Match server.
+func New() *Server {
+	return &Server{}
+}
+
+func startServingIndefinitely(params *Params) (func(), func(), error) {
+	s := New()
+
+	// Start serving traffic.
+	waitForServing, err := s.Start(params)
+	if err != nil {
+		serverLogger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Fatal("Failed to start gRPC and HTTP servers.")
+		return func() {}, func() {}, err
+	}
+	serverLogger.Info("Server has started.")
+	// Exit when we see a signal
+	waitUntilKilled, stopServingFunc := signal.New()
```

Comment:
Rename to waitForTerminateSignalFunc and sendTerminateSignalFunc? I think stopServingFunc is not a good representation of what signal.New() does.

Reply:
I named the variable in terms of it's intent vs what it actually does.